Add ability to redirect to/from homepage by using “/”

mobinzk opened the pull request:
- Bump version to 0.6.0-rc1.
- Add tenant URL to the from field.
- From url field now accepts “/“ to redirect to hompage and both “/foo“ and “foo” are acceptable.
- To url field now accepts “/” to redirect to homepage as well as “foo” or “/foo“ or external URLs

peterkeating commented on Redirects/Routing/RedirectRouteTransformer.cs:
```diff
@@ -33,5 +34,22 @@ public override ValueTask<RouteValueDictionary> TransformAsync(HttpContext httpC
+
+        private string GetContentItemId(HttpContext httpContext)
+        {
+            // Homepage URL
```
Not sure why this comment is here? It isn't the homepage URL is it? Looks like the URL for the current request.

peterkeating commented on Etch.OrchardCore.SEO.csproj:
```diff
@@ -4,7 +4,7 @@
     <AddRazorSupportForMvc>true</AddRazorSupportForMvc>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-    <Version>0.5.1-rc1</Version>
+    <Version>0.6.0-rc1</Version>
```
I would actually say this should be `0.5.2-rc1`. There aren't any breaking changes for `0.5.x` with this update, so incrementing the patch number feels more appropriate.

mobinzk replied:
I thought the first number `0(<- this one).0.0` is for when we update to a breaking change. Will update.

peterkeating replied:
@mobinzk I stand corrected, you're right. I do see this as a bug fix though and not additional functionality. It should handle redirecting from/to "/".

peterkeating commented on Redirects/Services/TenantService.cs:
```diff
@@ -0,0 +1,48 @@
+        {
+            var requestHostInfo = _httpContextAccessor.HttpContext.Request.Host;
+
+            var tenantUrlHost = _currentShellSettings.RequestUrlHost?.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault() ?? requestHostInfo.Host;
```
Not sure why there is a link break here? I'd have the two variable declarations grouped together and then have a line between the last variable declaration and the `if` statement.

peterkeating commented on Controllers/RedirectController.cs:
```diff
@@ -37,6 +41,11 @@ public async Task<IActionResult> Index(string contentItemId)
 
+            if (part.ToUrl.StartsWith("/", System.StringComparison.Ordinal))
+            {
+                return new RedirectResult($"{_tenantService.GetTenantUrl()}{part.ToUrl.TrimStart('/')}", part.IsPermanent);
```
The `GetTenantUrl` returns a URL without a trailing `/` and any prefixed `/` is being stripped so this causes the redirect URL to be incorrect. For example, if my tenant URL prefix was "example" and I was redirecting to "/search", the redirect URL would become "/examplesearch" instead of "/example/search".

peterkeating reviewed (changes requested):
Couldn't get redirecting from the homepage to another page to not work. Tried deleting the homepage content item to ensure there wasn't another page that was set to be the homepage but still no luck.

mobinzk commented:
As you mentioned when user has set a content item to homepage that would prevent redirect to work. The only solution Ii found so far is to disable HomeRoute module.

peterkeating commented:
@mobinzk If that's the case then add that is a "Known Issue" within the Redirects section in the README (e.g. https://github.com/EtchUK/Etch.OrchardCore.SEO#known-issues).

mobinzk committed:
Add ability to redirect to/from homepage by using “/”

- Bump version to 0.5.2-rc1.
- Add tenant URL to the from field.
- From url field now accepts “/“ to redirect to hompage and both “/foo“ and “foo” are acceptable.
- To url field now accepts “/” to redirect to homepage as well as “foo” or “/foo“ or external URLs
- Add instructions for user to disable Home Route module when redirecting from homepage

mobinzk commented:
That should be done now.

peterkeating reviewed (approved):
Nice, work around worked nicely 👍
